### ISSUE TITLE
Getting a "Retrieval Failed! Error 3" on some games.

### ISSUE THREAD

**Issue by gitscosh** [Twombs/GOGcli-GUI#1]
This happens when I'm adding games to The Manifest.

Sometimes, I can try again and it'll work, but there are a handful of games that just won't add to the manifest. What is Error 3?

The following games are affected:

Breach & Clear
EARTHLOCK: Festival of Magic
EARTHLOCK
Riddle of the Sphinx: The Awakening
RiME
The Witcher 3: Wild Hunt - Game of the Year Edition

I've got 924 games and these are the stubborn ones. Any ideas?

**Comment by Twombs** [comment]
There is something going on with GOG and their SDK at the moment.
It may be temporary and be restored after the current sale or not.
If not, then we will need to wait for Magnitus to make fixes to gogcli.exe.
I suggest you keep trying or wait for a resolution.
All the other downloaders seem to be impacted by this, so not just gogcli.exe, and perhaps even Galaxy.
There is a long discussion about the issues etc at the following, in which Magnitus also participates.
https://www.gog.com/forum/general/gogrepopy_python_script_for_regularly_backing_up_your_purchased_gog_collection_for_full_offline_e

**Comment by Twombs** [comment]
When I get a chance, I will look into error 3, but right now I suspect it is all the same issue others are experiencing. Magnitus has certainly experienced the same issue as others, though I am yet to personally.

**Comment by gitscosh** [comment]
Ok, thank you. Yes, other downloaders as well. I decided to try your app (and Magnitus's app) since I was having problems with gogrepoc.py giving me "OverflowError: size does not fit in an int" errors. It's a little frustrating. Hopefully, GOG will resolve it soon.

**Comment by BloodShed-Oni** [comment]
I'm getting it too, but only after being forced to close the command prompt because it's running a loop with this message over and over.

`[sdk] GetDownloadFileInfo(downloadPath=/downloads/metal_slug/76003) -> body retrieval error: stream error: stream ID 13; PROTOCOL_ERROR; received from peer. Will retry.`

The ID seems to vary (I've seen 13, 7 and 1).

**Comment by Twombs** [comment]
Yep, gogcli.exe keeps trying for quite a while in some cases. Not anything I can do about that, it is all down to how Magnitus has coded his program.

**Comment by Twombs** [comment]
Today using the latest version of gogcli.exe I was able to get the manifest entries for the few troublesome games that were still outstanding for me. It is not a quick process, and Dying Light: The Following took around 25 minutes to complete for me, but mileage will likely vary depending on PC and web connection. I have virtually all the DLCs for Dying Light, so a lot to process I guess. So whatever work-around Magnitus has done, appears to work, but very slowly.

**Comment by Gaerthal-Estelmist** [comment]
Greetings, I've encountered this error today after installing v3.4 . I'm not sure if it was there before v3.4 .
Like many other games when I try to add Champions of Krynn to the manifest, for example, everything is fine.
But when I try to add Curse of the Azure Bonds, I get 
[sdk] getDownloadFileInfo(downloadPath=/downloads/curse_of_the_azure_bonds/59963) -> body retrieval error: stream error: stream ID 1; PROTOCOL_ERROR; received from peer. Will retry.

As BloodShed-Oni wrote, the ID in the message varies. Thanks for this great tool!

**Comment by Twombs** [comment]
Sorry for the late reply. I keep forgetting to check my Gmail account, which used to be automatically checked by Thunderbird, until Google changed something. Best place to contact me is at GOG.

What version of gogcli.exe are you using? I recommend using the latest, which solved all these issues for me.

**Comment by Laxii-null** [comment]
It also happends to me for example with Absolute Drift or with Worms Armageddon right now.
